### PR TITLE
Implement search UI

### PR DESCRIPTION
**Pull request by jonathansick**
This PR implements a search UI with [Algolia instantsearch](https://www.algolia.com/doc/guides/building-search-ui/what-is-instantsearch/react/).

This work is built around the concept that content and search results should be shown directly on the homepage rather than in per-category listing pages. To do this I've pared down the UI so we don't have the category cards and large search box on the homepage.

This replaces #22 which was built around the idea of per-category result pages.

A demo of the design is available at https://astropy-search-demo.jsick.dev/

### CODE REVIEW

**Comment by jonathansick** [comment]
This should be merged _after_ #24

**Comment by jonathansick** [comment]
@adrn There's still more to do but this is 80% of it, and it might be easier to make smaller improvements in individual PRs after this.

**Comment by adrn** [comment]
Awesome!! I'll take a look over the next few days.

File: src/components/footer.js [hunk 1]
```diff
@@ -2,39 +2,125 @@ import React from 'react';
...
+                Learn more.
+              </a>
+            </p>
+          </article>
+          <article className="sponsors">
+            <h2>Sponsors</h2>
+            <p>
+              The Learn Astropy website was developed under a grant by the{' '}
+              <a href="http://www.dunlap.utoronto.ca">
+                Universitory of Toronto Dunlap Institute for Astrophysics.
```
Comment (adrn) [inline src/components/footer.js:106]:
```suggestion
                University of Toronto Dunlap Institute for Astrophysics.
```

File: src/components/footer.js [hunk 2]
```diff
@@ -2,39 +2,125 @@ import React from 'react';
...
+              </a>{' '}
+            </p>
+            <div className="numfocusStamp">
+              <a href="https://numfocus.org">
+                <img
+                  className="numfocusStamp__image"
+                  src={numfocusStamp}
+                  alt="A fiscally sponsored project of Numfocus"
```
Comment (adrn) [inline src/components/footer.js:114]:
```suggestion
                  alt="A fiscally sponsored project of NumFOCUS"
```

File: src/searchClient.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import algoliasearch from 'algoliasearch/lite';
+
+// This is the Search-only API key
+const searchClient = algoliasearch(
+  'KSDFJKHCO2',
+  'fadf7c74af324735d9e45d68481531ab'
```
Comment (adrn) [inline src/searchClient.js:6]: Is it ok for this to be public?

**Review by adrn — COMMENTED** [review]
Only a few minor comments! I didn't review the CSS closely but the styling in your demo site looks good to me :)

**Comment by eblur** [comment]
This looks good to me! I like that each card has a "Guide", "Tutorial", etc attached to it. I'm glad that they are labeled in that way.

One issue I note is that there is probably a lot of metadata still missing in order to properly search and sort. For example, the CCD Data Reduction Guide relies heavily on `ccdproc`, a package that is being developed to replace IRAF and will be of **key** interest to many people.

**Comment by eblur** [comment]
I would not require my above note to be fixed to accept the PR. After some discussion, we noticed it is related to astropy-librarian [Issue 14](https://github.com/astropy/learn-astropy-librarian/issues/14)

File: src/searchClient.js (hunk 3 above)
Comment (jonathansick) [reply src/searchClient.js:6]: Yeah, they're meant to be on the client-side and are read-only credentials.

**Comment by jonathansick** [comment]
@eblur That's a really good point about the keyword metadata for guides. We'll need to either establish a metadata file to go along with a guide to provide that or start extracting Python package usage from code samples as discussed in https://github.com/astropy/learn-astropy-librarian/issues/14 (which has some up-front challenges, but would be a nice system to have in the long run. Either way, the front-end wouldn't need to change to make that possible, so it's something I'll need to add to https://github.com/astropy/learn-astropy-librarian. I do think a metadata header standard or metadata file would be good for guides, and I've created a second ticket for that: https://github.com/astropy/learn-astropy-librarian/issues/17

**Pull request merged**